src/components/TodoList.tsx

```diff
@@ -0,0 +1,99 @@
+/* eslint-disable jsx-a11y/label-has-associated-control */
+/* eslint-disable jsx-a11y/control-has-associated-label */
+
+import { useContext, useEffect, useRef, useState } from 'react';
+import { DispatchContext, StatesContext } from '../context/Store';
+import classNames from 'classnames';
+import { TodoLoader } from './TodoLoader';
+
+export const TodoList: React.FC = () => {
+  const { todos, selectedTodo, filter } = useContext(StatesContext);
+  const dispatch = useContext(DispatchContext);
+  const [isEditing, setIsEditing] = useState(false);
+
+  const inputRef = useRef<HTMLInputElement>(null);
+  const filteredTodos = todos.filter(t => {
```

Comment:
You can wrap this _filteredTodos_ with an _useMemo_ hook with _filter_ as dependency, it will improve your code and prevent unnecessary re-renderings